Python/wrfrun/unCMIP6.py

```diff
@@ -0,0 +1,622 @@
+        c = 0 # Counter.
+        for idx, itm in self.vtable.iterrows():
+            # Var name and other info
+            varname = itm['src_v']
+            outvarname = itm['aim_v']
+            lvltype = itm['type']
+            lvlmark = itm['lvlmark']
+            frq = itm['freq']            
+            if lvlmark == 'None':
+                lvlmark = ''
+            # File name
+            fn = input_root+'/'+varname+'_'+frq+lvlmark+'_'+self.model_name
+            fn = fn+'_'+self.exp_id+'_'+self.esm_flag+'_'+self.grid_flag
+            if self.filestrdates[c]!='':
+                fn = fn+'_'+self.filestrdates[c]+'.nc'
+            else:    
+                fn = fn+'.nc'  
+            # Open dataset
+            ds = xr.open_dataset(fn)
+            # lat and lon handeling
+            if ((self.lat==None).all() and (self.lon==None).all()):
+                if ((varname=='tos') or (varname=='siconc') or (varname=='sithick')):
+                    raise ValueError("The first field in the table should be a variable different than tos, siconc or sithick.")
+                self.lat = np.array(ds.lat.values)
+                self.lon = np.array(ds.lon.values)
+            else:
+                if ((varname!='tos') and (varname!='sithick') and (varname!='siconc')):
+                    if not((np.allclose(np.array(ds.lat.values),self.lat,rtol=0.0,atol=1.0e-12)) and \
+                        (np.allclose(np.array(ds.lon.values),self.lon,rtol=0.0,atol=1.0e-12))):
+                        raise ValueError("Error: Inconsistent lat/lon values between different fields.")
+            # plev handeling for 3D fields
+            if lvltype=='3d':
+                if ((self.plev==None).all()):
+                    if (varname=='ua') or (varname=='va'):
+                        raise ValueError("The first 3D field in the table should be a variable different than ua and va.")
+                    self.plev = np.array(ds.plev.values)
+                else:
+                    if not(np.allclose(np.array(ds.plev.values),self.plev,rtol=0.0,atol=1.0e-12)):
+                        raise ValueError("Error: Inconsistent pressure levels between 3D fields.")            
+            # Find the date to read
+            if (self.filestrdates[c]!=''):
+                avail_times_t = ds[varname].time.values
+                avail_times = [pd.to_datetime(ele) for ele in avail_times_t]
+                if itm['approx_dates']==False: 
+                    if self.input_date in avail_times:
+                        if avail_times.count(self.input_date)!=1:
+                            raise ValueError('Error: Number of times that date/time appears in file is not 1.')
+                        cdate = self.input_date
+                    else:
+                        raise ValueError('Error: Could not find the exact date and time in the file.')
+                else:
+                    if (not(sorted(avail_times)==avail_times)):
```

Comment:
I don't think you need to check this...

Reply:
We talked about this. This whole code part is now gone.